source/PlayerInfo.cpp

```diff
@@ -658,7 +658,12 @@ void PlayerInfo::Die(int response, const shared_ptr<Ship> &capturer)
 	isDead = true;
 	// The player loses access to all their ships if they die on a planet.
 	if(GetPlanet() || !flagship)
+	{
+		// Zero out the flagship's velocity to prevent camera drift.
+		if(flagship)
+			flagship.get()->SetVelocity(Point());
```

Comment:
Is this not already covered by these lines in Engine? If you're on a planet, the flagship's velocity shouldn't be effecting the camera.
https://github.com/endless-sky/endless-sky/blob/31140ccb77b2b572d6d42c5d220ff89371bb625f/source/Engine.cpp#L504-L506

Edit: Oh, nvm. Dying sets your planet to null.

https://github.com/endless-sky/endless-sky/blob/f7d58797265c1962272afe5829afdb168bc30c05/source/PlanetPanel.cpp#L82-L87

Reply:
The player's planet is unset as a result of the death code in PlanetPanel, meaning there is no object to snap to.

Reply:
Perhaps instead of this, the Engine should change this line:
https://github.com/endless-sky/endless-sky/blob/31140ccb77b2b572d6d42c5d220ff89371bb625f/source/Engine.cpp#L507
to `else if(flagship && !player.IsDead())`

Although if we're clearing your ships, you should already have no flagship, and it should drop down to snapping to the current position. 🤔

Reply:
The player dies as soon as their flagship's hull reaches zero; this would make the camera immediately freeze instead of following an exploding flagship.

Reply:
Then might it make more sense to clear/reset the flagship pointer instead of altering the flagship in this case?

Reply:
Clearing the flagship pointer results in #10161 occurring if you die mid-flight, or the game outright crashing if you die while landed.

Reply:
Understandable. Have a nice day. :+1: